src/github.ts

```diff
@@ -0,0 +1,140 @@
+import * as github from '@actions/github';
+import * as core from '@actions/core';
+import { components } from '@octokit/openapi-types';
+
+const token = core.getInput('token', { required: true });
+const client = github.getOctokit(token);
+
+/**
+ * Get the current context's PR information
+ */
+export function getPr(): components['schemas']['pull-request-minimal'] {
+  const pullRequest = github.context.payload.workflow_run.pull_requests[0];
+  if (!pullRequest) {
+    core.debug(`NO PR FOUND IN CONTEXT (github.content.payload.workflow_run.pull_requests[0]):`);
+    core.setFailed(`PR is undefined. Exiting.`);
+    core.endGroup(); // Gathering inputs...
+    process.exit(1);
+  }
+
+  return pullRequest;
+}
+
+/**
+ * Request reviews on the current context's PR from the specified reviewers.
+ * @param reviewers The list of reviewers to request a review.
+ */
+export async function requestReviewers(reviewers: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Requesting Reviewers for PR #${pr} to: ${reviewers.join(',')}`);
+
+  const response = await client.rest.pulls.requestReviewers({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
+    pull_number: pr,
+    reviewers: reviewers,
+  });
+  coreDebugJson(response, `requestReviewers(${pr}, [${reviewers.join(',')}]) > response`);
+}
+
+/**
+ *
+ * Assign the current context's PR to the specified assignees.
+ * @param assignees The list of assignees to assign the PR.
+ */
+export async function setAssignees(assignees: string[]): Promise<void> {
+  const pr = getPr().number;
+  core.info(`Setting Assignees for PR #${pr} to: ${assignees.join(',')}`);
+
+  const response = await client.rest.issues.addAssignees({
+    owner: github.context.repo.owner,
+    repo: github.context.repo.repo,
```

Comment:
Why not destructure these like you do on line 65 below?

Reply:
Fixed.

No particular reason - this was just a miss on my part; thanks for catching! (related; is there any linting that will catch this sort of thing?)